.github/workflows/ci.yml

```diff
@@ -27,3 +27,19 @@ jobs:
 
       - name: Run tests
         run: npm test
+
+      - name: Generate README
+        uses: docker://pandoc/core:2.18
+        with:
+          entrypoint: ./docs/generate
+
+      - name: Check generated README is up to date
+        run: |
+          if [ -n "$(git status --porcelain README.md)" ]; then
+            echo 'There is a mismatch between the autogenerated `README.md` file and the source files in the `docs` directory. If you edited `README.md` directly, please change the files in `docs` instead, then run `./docs/generate`. If you already modified the files in `docs`, please run `./docs/generate`.'
+            echo 'Note that you need to have Pandoc (https://pandoc.org) installed to run `./docs/generate`.'
+            echo
+            echo "Here's the diff:"
```

Comment:
Do you want to suggest any improvements to this error message?

Also, maybe we should mention the Pandoc version here somewhere?

Reply:
The error message is great, thanks.

We could maybe tell users to look up the currently used Pandoc version in `.github/workflows/ci.yml`, yes.